Snowdog_SolrNavigation/web/js/solr-add-to-cart.js

```diff
@@ -0,0 +1,143 @@
+        data: form.serialize(),
+        type: 'post',
+        dataType: 'json',
+
+        beforeSend: function () {
+          if (self.isLoaderEnabled()) {
+            $('body').trigger(self.options.processStart);
+          }
+        },
+
+        success: function (res) {
+          var eventData,
+              parameters;
+
+          $(document).trigger('ajax:addToCart', {
+            'form': form,
+            'response': res
+          });
+
+          if (self.isLoaderEnabled()) {
+            $('body').trigger(self.options.processStop);
+          }
+
+          if (res.backUrl) {
+            eventData = {
+              'form': form,
+              'redirectParameters': []
+            };
+            // trigger global event, so other modules will be able add parameters to redirect url
+            $('body').trigger('catalogCategoryAddToCartRedirect', eventData);
+
+            if (eventData.redirectParameters.length > 0) {
+              parameters = res.backUrl.split('#');
+              parameters.push(eventData.redirectParameters.join('&'));
+              res.backUrl = parameters.join('#');
+            }
+            window.location = res.backUrl;
+
+            return;
+          }
+
+          if (res.messages) {
+            $(self.options.messagesSelector).html(res.messages);
+          }
+
+          if (res.minicart) {
+            $(self.options.minicartSelector).replaceWith(res.minicart);
+            $(self.options.minicartSelector).trigger('contentUpdated');
+          }
+
+          if (res.product && res.product.statusText) {
+            $(self.options.productStatusSelector)
+              .removeClass('available')
+              .addClass('unavailable')
+              .find('span')
+              .html(res.product.statusText);
+          }
+          self.enableAddToCartButton(form);
+        }
+      });
+    },
+
+    disableAddToCartButton: function (form) {
+      var addToCartButtonTextWhileAdding = $t('Adding...'),
+          addToCartButton                = $(form).find('button[type=submit]');
+
+      addToCartButton.prop('disabled', true);
+      addToCartButton.addClass('disabled');
+      addToCartButton.find('span').text(addToCartButtonTextWhileAdding);
+      addToCartButton.attr('title', addToCartButtonTextWhileAdding);
+    },
+
+    enableAddToCartButton: function (form) {
+      var addToCartButtonTextAdded = $t('Added'),
+          addToCartButton          = $(form).find('button[type=submit]');
+
+      addToCartButton.find('span').text(addToCartButtonTextAdded);
+      addToCartButton.attr('title', addToCartButtonTextAdded);
+
+      setTimeout(function () {
+        var addToCartButtonTextDefault = $t('Add to Cart');
+
+        addToCartButton.prop('disabled', false);
+        addToCartButton.removeClass('disabled');
+        addToCartButton.find('span').text(addToCartButtonTextDefault);
+        addToCartButton.attr('title', addToCartButtonTextDefault);
+      }, 2500);
```

Comment:
why 2500?

Reply:
For two reasons:
- Protect server overload (You can spam click on active button and make a lot of unecessary ajax request)
- UX - IMO User should receive a longer message b/c right it's chaning like:
* "Adding" - 1 second
* "Added" - 1 second (increased to 2.5second)